### DISCUSSION TITLE
Using "family" for chat rooms by their IDs

### DISCUSSION THREAD

**Discussion by meroo36** [rrousselGit/riverpod#3042] [Q&A] [answered]
I am implementing a chat app(like instagram's). I didnt want to have a single provider that stores rooms and chats. My idea was to use family. Eg:

```dart
final chatRoomProvider =
    StateNotifierProvider.family<ChatRoomProviderNotifier, ChatRoomState, String>((ref, roomId) {
  return ChatRoomProviderNotifier(ref, roomId);
});
```

with this idea i will have different providers for each room by their IDs. Is this some kind of anti-pattern or bad usage of families?

any idea or input is appreciated.

**Comment by rrousselGit** [comment, marked as answer]
That's not anti-pattern at all. That's typically the sort of use-case for its existence.

Although I would heavily suggest making your provider "autoDispose" too, to avoid memory leak when an ID stops being used
